Lesson16_Task01.py

```diff
@@ -0,0 +1,136 @@
+"""
+Home-task 13 Class writing Practice
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes, as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters
+# do not forget about block '''if __name__ == '__main__':'''
+"""
+
+import datetime
+
+
+class BankCard:
+    """
+    This class represents a bank card with various attributes and operations.
+    """
+    PIN_SIZE = 4  # Example of a class attribute
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date):
+        self.card_num = card_num
+        self._cvv = cvv  # Protected attribute
+        self._pin = pin  # Protected attribute
+        self.name = name
+        self.surname = surname
+        self.end_date = end_date
+        self.is_blocked = False
+        self.balance = 0.0
+
+    @staticmethod
+    def generate_expiry_date():
+        """
+        Generate an expiry date for the card.
+        """
+        today = datetime.date.today()
+        expiry_date = today.replace(year=today.year + 5)
+        return expiry_date
+
+    @property
```

Comment:
add please @property.setter for cvv and pin  - in real life user able to not only to read them but modify too
This will automatically allow you to remove set_pin method above. No sense invent methods with different names. 
@property and @property.setter allows you use single name for get and set operations. Check lesson examples.